### PR TITLE
cmd/prof: raise error when cmd is bash file

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----
`brew prof update` caused an error because `update` is made by bash, not ruby. However, the error message `Error: Unknown command: update` was not correct, so I added the correct message.

```bash
brew prof update
Error: Unknown command: update
Error: Failure while executing; `/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby /usr/local/Homebrew/Library/Homebrew/brew.rb update` exited with 1.
```

### CODE REVIEW

File: Library/Homebrew/commands.rb [hunk 1]
```diff
@@ -197,4 +197,8 @@ def rebuild_commands_completion_list
     file = HOMEBREW_CACHE/"all_commands_list.txt"
     file.atomic_write("#{commands(aliases: true).sort.join("\n")}\n")
   end
+
+  def only_bash_command_list
+    internal_commands.reject { |cmd| valid_internal_cmd?(cmd) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/commands.rb:202]: I think this function makes sense but not the implementation. This should check for a file extension instead.

File: Library/Homebrew/brew.rb [hunk 2]
```diff
@@ -131,6 +131,8 @@ class MissingEnvironmentVariables < RuntimeError; end
     end
     exec "brew-#{cmd}", *ARGV
   else
+    raise "command made by bash not ruby: #{cmd}" if Commands.only_bash_command_list.include?(cmd)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/brew.rb:134]: It doesn't seem like this should be in `brew.rb` but `prof.rb` instead.

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks @hyuraku! some comments.

File: Library/Homebrew/dev-cmd/prof.rb [hunk 3]
```diff
@@ -26,6 +26,8 @@ def prof
 
     brew_rb = (HOMEBREW_LIBRARY_PATH/"brew.rb").resolved_path
     FileUtils.mkdir_p "prof"
+    cmd = args.named.first
+    raise "#{cmd} is made by bash not ruby" if Commands.path(cmd).extname == ".sh"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/prof.rb:30]:
```suggestion
    raise UsageError, "#{cmd} is a Bash command!" if Commands.path(cmd).extname == ".sh"
```

**Comment by hyuraku** [comment]
@MikeMcQuaid 
Could you review this PR again?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Pull request merged**